mysql/datadog_checks/mysql/statement_samples.py

```diff
@@ -228,6 +228,7 @@ def __init__(self, check, config, connection_args):
             'events_statements_row_limit', 5000
         )
         self._explain_procedure = self._config.statement_samples_config.get('explain_procedure', 'explain_statement')
+        self._no_explain_reason = ''
```

Comment:
Why is this variable a class attribute? It should only be a local variable,  otherwise we can get some nasty bugs and race conditions.

Reply:
Yeah, the intent behind it was to make it easy/cleaner (imo) to set the reason for each statement rather than returning local variables around the possible failure locations. But you're totally right about the possible race conditions 👍.